### PR TITLE
add evaluate job

### PR DESCRIPTION
**Pull request by heluocs**
support model evaluate job

### CODE REVIEW

File: pkg/util/kubectl/kubectl.go [hunk 1]
```diff
@@ -284,6 +284,7 @@ func SaveAppConfigMapToFile(name, key, namespace string) (fileName string, err e
 	fmt.Printf("%s", string(out))
 
 	if err != nil {
+		log.Errorf("====== error: %v", err)
```
Comment (happy2048) [inline pkg/util/kubectl/kubectl.go:287]: please use log.Debugf

File: pkg/argsbuilder/evaluatejob.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	command.Flags().StringVar(&e.args.ModelPath, "model-path", "", "the model path to evaluate in the container")
+	command.Flags().StringVar(&e.args.DatasetPath, "dataset-path", "", "the model version to evaluate")
+	command.Flags().StringVar(&e.args.MetricsPath, "metrics-path", "", "the evaluate result saved path")
+	command.Flags().StringVar(&e.args.Image, "image", "", "the evaluate image")
+	command.Flags().StringVar(&e.args.WorkingDir, "working-dir", "/root", "working directory to extract the code. If using syncMode, the $workingDir/code contains the code")
+	command.Flags().IntVar(&e.args.GPUCount, "gpus", 0, "the limit GPU count of each replica to run the evaluate job.")
+	command.Flags().StringVar(&e.args.Cpu, "cpu", "", "the request cpu of each replica to run the evaluate job.")
+	command.Flags().StringVar(&e.args.Memory, "memory", "", "the request memory of each replica to run the evaluate job.")
+
+	command.Flags().StringSliceVarP(&envs, "env", "e", []string{}, "the environment variables")
```
Comment (happy2048) [inline pkg/argsbuilder/evaluatejob.go:84]: please replace StringSliceVarP to StringArrayVarP

File: pkg/argsbuilder/evaluatejob.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	command.Flags().StringSliceVarP(&annotations, "annotation", "a", []string{}, "the annotations")
```
Comment (happy2048) [inline pkg/argsbuilder/evaluatejob.go:85]: please replace StringSliceVarP to StringArrayVarP

File: pkg/argsbuilder/evaluatejob.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+	command.Flags().StringSliceVarP(&labels, "label", "", []string{}, "the labels")
```
Comment (happy2048) [inline pkg/argsbuilder/evaluatejob.go:86]: please replace StringSliceVarP to StringArrayVarP

File: pkg/util/kubectl/kubectl.go (hunk 1 above)
Comment (heluocs) [reply pkg/util/kubectl/kubectl.go]: delete the  log

File: pkg/argsbuilder/evaluatejob.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+	}
+	dataDirs = value.(*[]string)
+	log.Debugf("dataDir: %v", *dataDirs)
+	for i, dataDir := range *dataDirs {
+		hostPath, containerPath, err := util.ParseDataDirRaw(dataDir)
+		if err != nil {
+			return err
+		}
+		e.args.DataDirs = append(e.args.DataDirs, types.DataDirVolume{
+			Name:          fmt.Sprintf("training-data-%d", i),
```
[the remark this answers is not in the document]
Comment (heluocs) [reply pkg/argsbuilder/evaluatejob.go]: fixed

File: pkg/argsbuilder/evaluatejob.go (hunk 2 above)
Comment (heluocs) [reply pkg/argsbuilder/evaluatejob.go]: fixed

File: pkg/argsbuilder/evaluatejob.go (hunk 3 above)
Comment (heluocs) [reply pkg/argsbuilder/evaluatejob.go]: fixed

File: pkg/argsbuilder/evaluatejob.go (hunk 4 above)
Comment (heluocs) [reply pkg/argsbuilder/evaluatejob.go]: fixed

**Pull request merged**